components/form/docs/form-element-mixin.md

```diff
@@ -0,0 +1,157 @@
+# FormElementMixin
+
+The `FormElementMixin` allows the user to turn a custom web component into a form element.
+
+This means that the component will be able to:
+1. Perform live self-validation when being edited.
+1. Participate in validation and submission when added to `d2l-form` or `d2l-form-native`.
+
+All custom form elements must provide a form value that will be submitted during `d2l-form` or `d2l-form-native` submission.
+However, participating in validation is optional. Some custom form elements may not need validation because they have no
+invalid state.
+
+## Form Value
+
+**Properties:**
+- `name` (String): The name of the form control. Submitted with the form as part of a name/value pair.
```

Comment:
nit: we're moving towards properties tables instead of lists (e.g., https://github.com/BrightspaceUI/core/tree/master/components/button#d2l-button)